src/Elemental/Form/Interaction.elm

```diff
@@ -1,101 +1,183 @@
 module Elemental.Form.Interaction exposing
```

Comment:
I think the API is fine but I had a suggestion for the implementation.

```elm
module Elemental.Form.Interaction exposing
    ( Interaction
    , and
    , none
    , onBlur
    , onClick
    , onDoubleClick
    -- , onFocus
    -- , onMouseDown
    -- , onMouseEnter
    -- , onMouseLeave
    -- , onMouseOut
    -- , onMouseOver
    -- , onMouseUp
    -- , onTouchEnd
    -- , onTouchMove
    -- , onTouchStart
    , toAttrs
    )


import Html as H
import Html.Events as HE


type Interaction msg =
    -- NOTES:
    --
    -- Sets would have been best but we can't use it for custom types.
    -- 
    -- Alternatives:
    --
    -- 1. We could use a third-party set library that doesn't have that
    --    restriction.
    -- 2. We can swap out (::), see onClick below, for an implementation of
    --    insert that overwrites duplicates.
    -- 3. Explicitly decide how we want to handle duplicate actions in the list.
    --   a. Use the first one.
    --   b. Use the last one. Which is what I think your current implementation
    --       does.
    --
    -- Currently, new actions are added to the head of the list. So,
    --
    -- a
    -- |> and b
    -- |> and c
    --
    -- leads to [c, b, a]. Then, toAttrs leads to the events [ec, eb, ea]. When
    -- that attribute list is used on an element it means ea will take
    -- precedence over ec, assuming a == c, i.e. the implementation does 3(a).
    -- If we prefer 3(b) then we just need to reverse the list before returning
    -- it from toAttrs.
    Interaction (List (Action msg) -> List (Action msg))


type Action msg
    = Click msg
    | DoubleClick msg
    -- | MouseEnter msg
    -- | MouseLeave msg
    -- | MouseOver msg
    -- | MouseOut msg
    -- | MouseDown msg
    -- | MouseUp msg
    -- | TouchStart msg
    -- | TouchEnd msg
    -- | TouchMove msg
    -- | Focus msg
    | Blur msg


none : Interaction msg
none =
    Interaction identity


onClick : msg -> Interaction msg
onClick =
    Interaction << (::) << Click


onDoubleClick : msg -> Interaction msg
onDoubleClick =
    Interaction << (::) << DoubleClick


-- ...


onBlur : msg -> Interaction msg
onBlur =
    Interaction << (::) << Blur


and : Interaction msg -> Interaction msg -> Interaction msg
and (Interaction b) (Interaction a) =
    Interaction (a >> b)


toAttrs : Interaction msg -> List (H.Attribute msg)
toAttrs (Interaction toActions) =
    List.map toEvent (toActions [])


toEvent : Action msg -> H.Attribute msg
toEvent action =
    case action of
        Click msg ->
            HE.onClick msg

        DoubleClick msg ->
            HE.onDoubleClick msg

        -- ...

        Blur msg ->
            HE.onBlur msg
```

Reply:
For 3(b) semantics we can do:

```elm
toAttrs : Interaction msg -> List (H.Attribute msg)
toAttrs (Interaction toActions) =
    toActions []
        |> List.map toEvent
        |> List.reverse
```

Reply:
> Interesting, I'll make the updates. Are we dropping some of the messages since they are unlikely to be useful for form interactions?

No, I just didn't add them all in the example. You can keep all the handlers you had.